bee-network/bee-autopeering/src/peer/stores/rocksdb.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::peer::{
+    lists::{ActivePeer, ActivePeersList, ReplacementPeersList},
+    peer_id::PeerId,
+    stores::PeerStore,
+    Peer,
+};
+
+use rocksdb::{AsColumnFamilyRef, DBWithThreadMode, IteratorMode, MultiThreaded, Options, WriteBatch};
+
+use std::{path::PathBuf, sync::Arc};
+
+const ACTIVE_PEERS_CF: &str = "active_peers";
+const REPLACEMENTS_CF: &str = "replacements";
+
+/// The config for the RocksDb peer store.
+#[derive(Clone)]
+pub struct RocksDbPeerStoreConfig {
+    path: PathBuf,
+    options: Options,
+}
+
+/// The (persistent) RocksDb peer store.
+#[derive(Clone)]
+pub struct RocksDbPeerStore {
+    db: Arc<DBWithThreadMode<MultiThreaded>>,
+    config: RocksDbPeerStoreConfig,
+}
+
+impl RocksDbPeerStore {
+    fn open_cf(&self, cf_str: &'static str) -> impl AsColumnFamilyRef + '_ {
+        self.db.cf_handle(cf_str).expect("error opening cf")
+    }
+}
+
+impl PeerStore for RocksDbPeerStore {
+    type Config = RocksDbPeerStoreConfig;
+
+    fn new(config: Self::Config) -> Self {
+        let db = DBWithThreadMode::open_cf(&config.options, &config.path, &[ACTIVE_PEERS_CF, REPLACEMENTS_CF])
+            .expect("error opening peer store");
+
+        Self {
+            db: Arc::new(db),
+            config,
+        }
+    }
+
+    fn store_active(&self, active_peer: ActivePeer) {
```

Comment:
I know the trait was already like that but I don't think we should panic all backend errors